presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilterConsumer.java

```diff
@@ -144,6 +144,11 @@ private void addPartition(TupleDomain<DynamicFilterId> tupleDomain)
             return buildChannels.keySet().stream()
                     .collect(toImmutableMap(identity(), filterId -> Domain.none(filterBuildTypes.get(filterId))));
         }
+        if (result.isAll()) {
+            // Set `all` domain explicitly for dynamic filters so that it can be waited for
+            return buildChannels.keySet().stream()
+                    .collect(toImmutableMap(identity(), filterId -> Domain.all(filterBuildTypes.get(filterId))));
```

Comment:
If you want to assume each `DynamicFilterId` has an entry in the Map, you would need to change the `return result.getDomains().get()` below too. However, it seems to be like having inexplicit assumptions in the data model, which are not necessary. Can we handle this in eg in `io.prestosql.operator.TaskContext#collectDynamicTupleDomain` instead?

Reply:
> If you want to assume each DynamicFilterId has an entry in the Map, you would need to change the return result.getDomains().get() below too. However, it seems to be like having inexplicit assumptions in the data model, which are not necessary.

What do you mean by `you would need to change the return result.getDomains().get() below too`. The only special cases for `TupleDomain` are `all` and `none`. For other cases there is an explicit mapping between filter id and domain.

> Can we handle this in eg in io.prestosql.operator.TaskContext#collectDynamicTupleDomain instead?

`TaskContext` doesn't know what DF there are.

Reply:
I've added assertion so that all expected DFs are present during `result.getDomains().get()` call

Reply:
The TupleDomain's map does not contain any Domain.all entries.
So TupleDomain.all (empty map) does not need to be treated as a special case.

Reply:
> So TupleDomain.all (empty map) does not need to be treated as a special case.

It does. Let's suppose there is listener for `DF1`. It might happen that build side is too large and `TupleDomain.all()` is reported. In that case we need to explicitly set `DF1` domain to `all`. Otherwise listener won't know that DF is available.

Reply:
@findepi I've updated the code. Indeed, `Domain.all` would be removed from `TupleDomain`. I've added support for that case along with test. However, in practice this wasn't happening because `DynamicFilterSourceOperator` did not return partial result.